Products/ZenModel/actions.py

```diff
@@ -394,76 +395,104 @@ def _encodeBody(self, body):
             plain_body = MIMEText(body.decode('ascii', 'ignore'))
         return plain_body
 
+    def _targetsByTz(self, dmd, targets):
+        """
+        Take timezone from user property to convert a event time in
+        notification and also group targets emails by those timezones.
+        """
+        targets_by_timezones = {}
+        for user in dmd.ZenUsers.getAllUserSettings():
+            if user.email in targets:
+                if targets_by_timezones.get(user.timezone):
+                    targets_by_timezones[user.timezone].append(user.email)
+                else:
+                    targets_by_timezones[user.timezone] = [user.email]
+        return targets_by_timezones
+
     def executeBatch(self, notification, signal, targets):
         log.debug("Executing %s action for targets: %s", self.name, targets)
         self.setupAction(notification.dmd)
+        targets_by_timezones = self._targetsByTz(notification.dmd, targets)
+        original_lst = signal.event.last_seen_time
+        original_fst = signal.event.first_seen_time
+        for target_timezone in targets_by_timezones:
+            # Convert timestamp to user timezone
+            signal.event.last_seen_time = long(
+                isoToTimestamp(convertTimestampToTimeZone(
+                    (original_lst / 1000),
+                    target_timezone, '%Y-%m-%d %H:%M:%S'))) * 1000
+            signal.event.first_seen_time = long(
+                isoToTimestamp(convertTimestampToTimeZone(
+                    (original_fst / 1000),
+                    target_timezone, '%Y-%m-%d %H:%M:%S'))) * 1000
+            targets = targets_by_timezones[target_timezone]
+
+            data = self._signalToContextDict(signal, notification)
+
+            # Check for email recipients in the event
+            details = data['evt'].details
+            mail_targets = details.get('recipients', '')
+            mail_targets = [x.strip() for x in mail_targets.split(',') if x.strip()]
+            if len(mail_targets) > 0:
+                log.debug("Adding recipients defined in the event %s", mail_targets)
+                targets |= set(mail_targets)
```

Comment:
Does this line, `targets |= set(mail_targets)`, work?

According to the `_targetsByTz` method, it looks like `targets` is a list and there is no `|=` operator defined for `list` and `set`.

You should add a test where there's an event having details containing a 'recipients' field.